+bisos/+iteration/+step/termination.m

```diff
@@ -0,0 +1,134 @@
+classdef termination < bisos.iteration.Step
+   
+properties
+    type = 'termination';
+    varout = [];
+    varin;
+
+    operator;
+    
+    % Structures with info about condition 
+    prev = struct('fhan', @(x) x, 'vars', []);
+    curr = struct('fhan', @(x) x, 'vars', []);
+end
+
+methods
+    function step = termination(~, previous, current, op, varargin)
+        % setup termination rule
+        
+        step.curr = step.setup2struct(step.curr, current);
+        step.varin = step.curr.vars;
+        step.prev = step.setup2struct(step.prev, previous);
+        
```

Comment:
we should check that the variable inputs to both `step.curr` and `step.prev` exist in the problem.

```suggestion
        cellfun(@(v) assert(hasvariable(prob,v), 'Unknown variable ''%s''.', v), [step.curr.vars step.prev.vars]);
```